app/Repositories/UserRepository.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace App\Repositories;
+
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Str;
+use Stripe\Customer;
+use TCG\Voyager\Models\Role;
+use Wave\User;
+
+class UserRepository
+{
+    public function getByEmail(string $email): ?User
+    {
+        return User::where('email', $email)->first();
+    }
+
+    public function updateStripeId(User $user, string $stripeId): void
+    {
+        $user->stripe_id = $stripeId;
+        $user->save();
+        $user->syncStripeCustomerDetails();
+    }
+
+    public function getUniqueUsernameFromEmail(string $email): string
+    {
+        $username = Str::before($email, '@');
+        $username_original = $username;
+        $counter = 1;
+
+        while ($this->getByUsername($username)) {
```

Comment:
Please restrain from performing sql requests in the loop. Maybe simply last name that starts with string $username and add a number to that?

Reply:
Updated